### PR TITLE
NumberFormatException and merge conflicts

### PR DESCRIPTION
**Pull request by SamHausmann**

### CODE REVIEW

File: api/src/main/java/com/basistech/rosette/api/RosetteAPI.java [hunk 1]
```diff
@@ -1753,6 +1758,12 @@ private String headerValueOrNull(Header header) {
             throws IOException, RosetteAPIException {
         int status = httpResponse.getStatusLine().getStatusCode();
         String encoding = headerValueOrNull(httpResponse.getFirstHeader(HttpHeaders.CONTENT_ENCODING));
+        try {
+            int concurrencyHeader = Integer.parseInt(headerValueOrNull(httpResponse.getFirstHeader("X-RosetteApi-Concurrency")));
```
Comment (guruxu) [inline api/src/main/java/com/basistech/rosette/api/RosetteAPI.java:1762]: what if the header is erroneously set to 0 or -1? unit tests?

Comment (SamHausmann) [inline api/src/main/java/com/basistech/rosette/api/RosetteAPI.java:1762]: By the server?

Comment (guruxu) [inline api/src/main/java/com/basistech/rosette/api/RosetteAPI.java:1762]: Yeah, don't trust the server will always do the right thing ...
Also no need to close PR to make changes, you can keep pushing and changes will show in the same PR.

File: api/src/main/java/com/basistech/rosette/api/RosetteAPI.java [hunk 2]
```diff
@@ -283,7 +285,10 @@ private void initHttpClient() {
         if (customHeaders.size() > 0) {
             defaultHeaders.addAll(customHeaders);
         }
-        httpClient = HttpClients.custom().setDefaultHeaders(defaultHeaders).build();
+
+        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager();
+        cm.setMaxTotal(maxSockets);
+        httpClient = HttpClients.custom().setConnectionManager(cm).setDefaultHeaders(defaultHeaders).build();
```
Comment (dgoldfarble) [inline api/src/main/java/com/basistech/rosette/api/RosetteAPI.java:291]: This code gets run once at initialization, and updating maxSockets doesn't update the connection manager.

File: api/src/main/java/com/basistech/rosette/api/RosetteAPI.java [hunk 3]
```diff
@@ -1753,6 +1758,12 @@ private String headerValueOrNull(Header header) {
             throws IOException, RosetteAPIException {
         int status = httpResponse.getStatusLine().getStatusCode();
         String encoding = headerValueOrNull(httpResponse.getFirstHeader(HttpHeaders.CONTENT_ENCODING));
+        try {
+            int concurrencyHeader = Integer.parseInt(headerValueOrNull(httpResponse.getFirstHeader("X-RosetteApi-Concurrency")));
+            maxSockets = concurrencyHeader;
+        } catch (NumberFormatException e) {
+            throw e;
```
Comment (dgoldfarble) [inline api/src/main/java/com/basistech/rosette/api/RosetteAPI.java:1765]: Is the server now required to set X-RosetteApi-Concurrency?
We should do something different with the exception than just throwing it back.

**Comment by dgoldfarble** [comment]
Moved to https://github.com/rosette-api/java/pull/59

**Pull request closed without merging**